### ISSUE TITLE
Compile error: ‘_rdseed64_step’ was not declared in this scope

### ISSUE THREAD

**Issue by define-null** [weidai11/cryptopp#140]
Hi, I get an error while trying to compile cryptopp (tag CRYPTOPP_5_6_3)

```
rdrand.cpp: In function ‘int CryptoPP::ALL_RSI_GenerateBlock(byte*, size_t, unsigned int)’:
rdrand.cpp:360:37: error: ‘_rdseed64_step’ was not declared in this scope
   if (_rdseed64_step((word64*)output))
                                     ^
rdrand.cpp:378:26: error: ‘_rdseed64_step’ was not declared in this scope
   if (_rdseed64_step(&val))
                          ^
make: *** [rdrand.o] Error 1
```
```
$ g++ --version
g++ (Ubuntu 4.8.4-2ubuntu1~14.04.1) 4.8.4

$ uname -a
Linux 4.2.0-25-generic #30~14.04.1-Ubuntu SMP Mon Jan 18 16:25:56 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```
I have found the same bug here https://www.spinics.net/lists/gcchelp/msg44760.html
The fix, that worked for me:

```
$ git diff --no-ext-diff
diff --git a/rdrand.h b/rdrand.h
index fc2b77b..8c9a95a 100644
--- a/rdrand.h
+++ b/rdrand.h
@@ -8,6 +8,7 @@
 #ifndef CRYPTOPP_RDRAND_H
 #define CRYPTOPP_RDRAND_H
 
+#include <x86intrin.h>
 #include "cryptlib.h"
 
 // This file (and friends) provides both RDRAND and RDSEED, but its somewhat
```

**Comment by noloader** [comment]
> The fix, that worked for me:
>  `+#include <x86intrin.h>`

Thanks. We encountered that too after 5.6.3 was released.

The problem was GCC does not provide documentation on RDRAND and RDSEED, and they told us to follow Intel's documentation. Intel puts the declaration in `<immintrin.h>` so that's what we used. We later learned that GCC does not always follow Intel's lead with respect to the instruction. Also see [Issue 68316: GCC C++ compiler cannot compile a program using RDESED intrinsics](http://gcc.gnu.org/ml/gcc-bugs/2015-11/msg01051.html).

The fix was committed in December 2015 at [951d8b9dbe37f46a](http://github.com/weidai11/cryptopp/commit/951d8b9dbe37f46a725bf9ed31315269a9703012).

-----

You might want to work out of Wei's GitHub to ensure you have the latest sources. All in all, Wei's GitHub is stable - we don't use it for experimentation. There's been a few minor bug fixes, so it will help you with issues like this and a few others like it. Also see [GitHub and Git](http://www.cryptopp.com/wiki/GitHub_and_Git) on the Crypto++ wiki.
